### PR TITLE
LPS-122492 - Replace next method from metal-dom

### PR DESCRIPTION
**Pull request by bryceosterhaus**
https://issues.liferay.com/browse/LPS-122466

Few notes:
1. `next` only seems to be used once in DXP
2. The function that uses `next` doesn't even seem to be used, but I figured its best to not remove that teams code in case they intend to use it.
3. Rather than use a relatively unknown api of `TreeWalker` I decided to use a similar implementation as metal-dom since its only used this once and was only a few lines of code.

Anyone have issues with this approach?

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/js/SiteNavigationMenuItemDOMHandler.js [hunk 1]
```diff
@@ -86,7 +85,19 @@ const getId = function (menuItem) {
...
 const getNextSibling = function (menuItem) {
-	next(menuItem, `.${MENU_ITEM_CLASSNAME}`);
+	const selector = `.${MENU_ITEM_CLASSNAME}`;
+
+	let element = menuItem;
+
+	while (element) {
+		element = element.nextSibling;
+
+		if (element && element.matches(selector)) {
```
Comment (wincent) [inline modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/js/SiteNavigationMenuItemDOMHandler.js:95]: If you want to check for the presence of a classname, it is probably better to use the more targeted API for it, and avoid the whole `selector` template literal.

```js
if (element && element.classList.has(MENU_ITEM_CLASSNAME))
```

We had [a semi-related discussion in the guidelines repo](https://github.com/liferay/liferay-frontend-guidelines/issues/19) about how best to select element and the consensus seemed to be that we should use more specific/targeted APIs when available, so I think the same applies here analogously.

Comment (bryceosterhaus) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/resources/META-INF/resources/js/SiteNavigationMenuItemDOMHandler.js:95]: 👍  good idea

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci...

**Comment by john-co** [comment]
ci:stop

**Comment by john-co** [comment]
ci:test:sf

**Comment by john-co** [comment]
ci:test:relevant

**Comment by john-co** [comment]
maybe ci just woke up from it's nap?...lol

**Comment by bryceosterhaus** [comment]
ci:test:sf

**Comment by bryceosterhaus** [comment]
ci:test:relevant

**Comment by bryceosterhaus** [comment]
ci:stop

**Comment by bryceosterhaus** [comment]
didn't see the previous test running... gonna stop and re-test for latest changes

**Comment by jbalsas** [comment]
ci:forward

**Comment by jbalsas** [comment]
Forwarding with @wincent's feedback applied and green CI!

**Pull request closed without merging**